src/kudu/consensus/consensus_queue.cc

```diff
@@ -1231,16 +1229,23 @@ int64_t PeerMessageQueue::ComputeNewWatermarkDynamicMode(int64_t* watermark) {
     // Refer to the comment in AdvanceQueueWatermark method for why only
     // successful last exchanges are considered.
     if (peer.second->last_exchange_status == PeerStatus::OK) {
-      const string& peer_region = peer.second->peer_pb.attrs().region();
-      if (peer_region == leader_region) {
-        watermarks_in_leader_region.push_back(
+      const string& peer_quorum_id = getQuorumId(peer.second->peer_pb);
+      if (peer_quorum_id == leader_quorum) {
+        watermarks_in_leader_quorum.push_back(
             peer.second->last_received.index());
       }
     }
   }
 
+  std::map<std::string, int> voter_distribution;
```

Comment:
I am concerned about lines 1240 - 1246. It seems to me we are directly
reading voter_distribution. Will this voter distribution be region based on
quorum id based. During rollouts this seems to me the same watermarking
code can use both types?

Are we recomputing voter_distribution during setting time as we initially discussed

Reply:
Sorry. This is a careless mistake from me. In my recent change, I reverted the changes in Static Mode so that Static Mode always uses region and original vd. I ended up mistakenly reverting VD in the Dynamic mode rather than Static Mode.